src/rgw/rgw_admin.cc

```diff
@@ -2639,6 +2642,234 @@ static int bucket_sync_status(rgw::sal::RGWRadosStore *store, const RGWBucketInf
   return 0;
 }
 
+template <typename T>
+int decode_system_obj(RGWSI_SysObj* svc, const rgw_raw_obj& obj, T& result)
+{
+  auto obj_ctx = svc->init_obj_ctx();
+  auto sysobj = svc->get_obj(obj_ctx, obj);
+  bufferlist bl;
+  int ret = sysobj.rop().read(&bl, null_yield);
+  if (ret < 0)
+    return ret;
+
+  try {
+    auto iter = bl.cbegin();
+    using ceph::decode;
+    decode(result, iter);
+    return 0;
+  } catch (const buffer::error& err) {
+    lderr(svc->ctx()) << "error decoding " << obj << ": " << err.what() << dendl;
+    return -EIO;
+  }
+}
+
+bool operator<(const std::vector<rgw_bucket_shard_sync_info>& lhs,
+               const BucketIndexShardsManager& rhs)
+{
+  for (size_t i = 0; i < lhs.size(); ++i) {
+    const auto& l = BucketIndexShardsManager::get_shard_marker(lhs[i].inc_marker.position);
+    const auto& r = rhs.get(i, "");
+    if (l >= r) {
```

Comment:
@cbodley I assume we have a few shards that are greater or equal and a others that are smaller, so maybe the logic needs to be reversed? That is if `(l < r)` return `true`.